java/dataset/src/main/java/org/apache/arrow/dataset/scanner/ScanOptions.java

```diff
@@ -69,4 +83,8 @@ public Optional<String[]> getColumns() {
   public long getBatchSize() {
     return batchSize;
   }
+
+  public ByteBuffer getSubstraitExtendedExpression() {
```

Comment:
Either both getters should use Optional or neither should.

Reply:
Added